library/core/src/ptr/non_null.rs

```diff
@@ -1803,4 +1816,44 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+   #[kani::proof_for_contract(NonNull::byte_sub)]
+    pub fn non_null_check_byte_sub() {
+    // Initializing array of size 10000
+    const ARR_SIZE: usize = 100000;
+    let arr: [i32; ARR_SIZE] = kani::any();
+    //Randomizing start pointer within array
+    let offset = kani::any_where(|x| *x <= ARR_SIZE);
+    // Create a NonNull pointer to the end of the array
+    let raw_ptr: *mut i32 = unsafe { arr.as_ptr().add(offset) as *mut i32 };
```

Comment:
After discussing with @zhassan-aws array indexing is not plausible because it only returns the value of the element and we cannot get the address of that element.

Reply:
@QinyuanWu I'm not sure why it didn't work yesterday when I tried it. But looks like we can do `&arr[offset] as *const i32`. For example, this works:
```rust
    let a = [1, 2, 3];
    let index: usize = 1;
    let p = &a[index] as *const i32;
    assert_eq!(unsafe { *p }, 2);
```

Reply:
Yes, what Zyad said above is what I meant!

Reply:
As @QinyuanWu noted in the following reply, it wouldn't be possible the recommended way.